### PR TITLE
PTX-2845 Adding first initial node affinity test

### PR DESCRIPTION
**Pull request by nikolaypopov**
Adding first initial node affinity test

Signed-off-by: nikolaypopov <nikolay.popov86@gmail.com>

### CODE REVIEW

File: test/integration_test/common_test.go [hunk 1]
```diff
@@ -9,18 +9,56 @@ import (
...
+	}
+
+	k8sOps := core.Instance()
+	k8sOps.SetConfig(config)
+
+	operatorOps := operator.Instance()
+	operatorOps.SetConfig(config)
+
+	return nil
+}
```
Comment (piyush-nimbalkar) [inline test/integration_test/common_test.go:60]: You don't need to do any of this. Just using `core.Instance()` or `operator.Instance()` anywhere in the code automatically initialize the kubernetes client to the current cluster. sched-ops handles that.
You need to call SetConfig only if you want to talk to a different Kubernetes cluster from this one. Stork has to perform migration tests, that's why it needs it. We don't need it here.

The assumption here is that we will be running the test pod in the same cluster as the operator. If you want to run the test locally, you can set the KUBECONFIG variable and it should talk to that k8s cluster.

File: test/integration_test/common_test.go [hunk 2]
```diff
@@ -9,18 +9,56 @@ import (
...
 
 const (
 	specDir = "./specs"
 )
 
 func TestMain(m *testing.M) {
+	if err := setup(); err != nil {
+		logrus.Errorf("Setup failed with error: %v", err)
+		os.Exit(1)
+	}
```
Comment (piyush-nimbalkar) [inline test/integration_test/common_test.go:28]: You can remove this as well, as there is no setup needed as of now. We can add it later if there is setup needed.

File: test/integration_test/node_affinity_test.go [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+			},
+		},
+	}
+
+	// Get K8S nodes
+	nodeList, err := core.Instance().GetNodes()
+	require.NoError(t, err)
+
+	// Set affinity Label one of the K8S nodes
+	for _, node := range nodeList.Items {
```
Comment (piyush-nimbalkar) [inline test/integration_test/node_affinity_test.go:62]: Make sure this is not a master node, else the test will fail. 
You can add this later when you add the validation for the test.

**Review by piyush-nimbalkar — APPROVED** [review]
LGTM after addressing the comments.

File: test/integration_test/common_test.go (hunk 1 above)
Comment (nikolaypopov) [reply test/integration_test/common_test.go]: Good to know, removed.

File: test/integration_test/common_test.go (hunk 2 above)
Comment (nikolaypopov) [reply test/integration_test/common_test.go]: Removed.

File: test/integration_test/node_affinity_test.go (hunk 3 above)
Comment (nikolaypopov) [reply test/integration_test/node_affinity_test.go:64]: Added check for master label.

File: test/integration_test/node_affinity_test.go [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+
+	// Get K8S nodes
+	nodeList, err := core.Instance().GetNodes()
+	require.NoError(t, err)
+
+	// Set affinity Label one of the K8S nodes
+	var nodeNameWithLabel string
+	for _, node := range nodeList.Items {
+		for key := range node.Labels {
+			if strings.Contains(key, "master") { // Skip master node, we don't need to label it
```
Comment (piyush-nimbalkar) [inline test/integration_test/node_affinity_test.go:66]: Use `node-role.kubernetes.io/master`

**Review by piyush-nimbalkar — APPROVED** [review]
Squash and merge your changes

File: test/integration_test/node_affinity_test.go (hunk 4 above)
Comment (nikolaypopov) [reply test/integration_test/node_affinity_test.go]: Done.

**Review by Rohit-PX — APPROVED** [review]

**Pull request merged**